Applications/SlicerApp/Testing/Python/CurvedPlanarReformation.py

```diff
@@ -0,0 +1,414 @@
+        # so that when the scene is saved and reloaded, these settings are restored.
+
+        self.setParameterNode(self.logic.getParameterNode())
+
+        # Select default input nodes if nothing is selected yet to save a few clicks for the user
+        if not self._parameterNode.inputVolume:
+            firstVolumeNode = slicer.mrmlScene.GetFirstNodeByClass("vtkMRMLScalarVolumeNode")
+            if firstVolumeNode:
+                self._parameterNode.inputVolume = firstVolumeNode
+
+    def setParameterNode(self, inputParameterNode: Optional[CurvedPlanarReformationParameterNode]) -> None:
+        """
+        Set and observe parameter node.
+        Observation is needed because when the parameter node is changed then the GUI must be updated immediately.
+        """
+
+        if self._parameterNode:
+            self._parameterNode.disconnectGui(self._parameterNodeGuiTag)
+            self.removeObserver(self._parameterNode, vtk.vtkCommand.ModifiedEvent, self._checkCanApply)
+        self._parameterNode = inputParameterNode
+        if self._parameterNode:
+            # Note: in the .ui file, a Qt dynamic property called "SlicerParameterName" is set on each
+            # ui element that needs connection.
+            self._parameterNodeGuiTag = self._parameterNode.connectGui(self.ui)
+            self.addObserver(self._parameterNode, vtk.vtkCommand.ModifiedEvent, self._checkCanApply)
+            self._checkCanApply()
+
+    def _checkCanApply(self, caller=None, event=None) -> None:
+        if self._parameterNode and self._parameterNode.inputVolume and self._parameterNode.thresholdedVolume:
+            self.ui.applyButton.toolTip = _("Compute output volume")
+            self.ui.applyButton.enabled = True
+        else:
+            self.ui.applyButton.toolTip = _("Select input and output volume nodes")
+            self.ui.applyButton.enabled = False
+
+    def onApplyButton(self) -> None:
+        """Run processing when user clicks "Apply" button."""
+        with slicer.util.tryWithErrorDisplay(_("Failed to compute results."), waitCursor=True):
+            # Compute output
+            self.logic.process(
+                self.ui.inputSelector.currentNode(),
+                self.ui.outputSelector.currentNode(),
+                self.ui.imageThresholdSliderWidget.value,
+                self.ui.invertOutputCheckBox.checked,
+            )
+
+            # Compute inverted output (if needed)
+            if self.ui.invertedOutputSelector.currentNode():
+                # If additional output volume is selected then result with inverted threshold is written there
+                self.logic.process(
+                    self.ui.inputSelector.currentNode(),
+                    self.ui.invertedOutputSelector.currentNode(),
+                    self.ui.imageThresholdSliderWidget.value,
+                    not self.ui.invertOutputCheckBox.checked,
+                    showResult=False,
+                )
```

Comment:
This should be simplified. There are no UI file and attempting to open the test module would fail.

```suggestion
    def __init__(self, parent=None) -> None:
        """Called when the user opens the module the first time and the widget is initialized."""
        ScriptedLoadableModuleWidget.__init__(self, parent)

    def setup(self) -> None:
        """Called when the user opens the module the first time and the widget is initialized."""
        ScriptedLoadableModuleWidget.setup(self)
```